config/src/config/storage_config.rs

```diff
@@ -8,6 +8,33 @@ use std::{
     path::PathBuf,
 };
 
+#[derive(Copy, Clone, Debug, Deserialize, PartialEq, Serialize)]
+#[serde(default, deny_unknown_fields)]
+pub struct RocksdbConfig {
+    /// Port selected RocksDB options for tuning underlying rocksdb instance.
+    /// see https://github.com/facebook/rocksdb/blob/master/include/rocksdb/options.h
+    /// for detailed explanations.
+    pub create_if_missing: bool,
+    pub create_if_missing_column_families: bool,
+    pub max_open_files: i32,
+    pub max_total_wal_size: u64,
+}
+
+impl Default for RocksdbConfig {
+    fn default() -> Self {
+        Self {
+            create_if_missing: true,
+            create_if_missing_column_families: true,
```

Comment:
When would these ever be changed?

Reply:
for val/fullnode, they don't. But for my tool to access db, or open_read_only(), I don't want to create a db if it doesn't exist there. It should be an error "DB not found"